src/yunohost/backup.py

```diff
@@ -199,257 +358,506 @@ def _clean_tmp_dir(retcode=0):
+            ret = subprocess.call(['umount', self.work_dir])
+            if ret == 0:
+                subprocess.call(['rmdir', self.work_dir])
+                logger.debug("Unmount dir: {}".format(self.work_dir))
             else:
-                if i == 'y' or i == 'Y':
-                    force = True
-            if not force:
-                _clean_tmp_dir()
-                raise MoulinetteError(errno.EEXIST, m18n.n('restore_failed'))
-    else:
-        # Retrieve the domain from the backup
-        try:
-            with open("%s/conf/ynh/current_host" % tmp_dir, 'r') as f:
-                domain = f.readline().rstrip()
-        except IOError:
-            logger.debug("unable to retrieve current_host from the backup",
-                         exc_info=1)
-            raise MoulinetteError(errno.EIO, m18n.n('backup_invalid_archive'))
+                raise MoulinetteError(errno.EIO,
+                                      m18n.n('restore_removing_tmp_dir_failed'))
+        elif os.path.isdir(self.work_dir):
+            logger.debug("temporary restore directory '%s' already exists",
+                         self.work_dir)
+            ret = subprocess.call(['rm', '-Rf', self.work_dir])
+            if ret == 0:
+                logger.debug("Delete dir: {}".format(self.work_dir))
+            else:
+                raise MoulinetteError(errno.EIO,
+                                      m18n.n('restore_removing_tmp_dir_failed'))
+
+        filesystem.mkdir(self.work_dir, parents=True)
+
+        self.method.mount(self)
 
-        logger.debug("executing the post-install...")
-        tools_postinstall(domain, 'yunohost', True)
+        self._read_info_files()
+
+    def _check_free_space(self, hooks, apps):
+        """ Check available disk space """
+        statvfs = os.statvfs(BACKUP_PATH)
+        free_space = statvfs.f_frsize * statvfs.f_bavail
+        needed_space = self.info['size'] + MARGIN_SPACE_SIZE * 1024 * 1024
+        if free_space < needed_space:
+
+            if hooks == [] and apps == []:
+                logger.debug("%dB left but %dB is needed", free_space,
+                             needed_space)
+                raise MoulinetteError(
+                    errno.EIO, m18n.n('not_enough_disk_space',
+                                      path=BACKUP_PATH))
+
+            # TODO partial restore don't need all backup size
+            # We should compute size for each apps and for each hooks
+            # how to know the dependencies size ?
+
+            # Different behaviour if it's only conf hook restoration
+            data_hooks = [hook for hook in hooks if hooks.startswith('data')]
+            if apps is not None or data_hooks != []:
+                logger.warning(m18n.n('may_be_not_enough_disk_space',
+                                      free_space=free_space,
+                                      needed_space=needed_space))
+            elif free_space < CONF_MARGIN_SPACE_SIZE * 1024 * 1024:
```

Comment:
CONF_MARGIN_SPACE_SIZE is not defined here ? Is this a typo for MARGIN_SPACE_SIZE ?